plugins/cost-insights/src/components/CostOverviewCard/CostOverviewByProduct.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import { useTheme, Box } from '@material-ui/core';
+import {
+  AreaChart,
+  ContentRenderer,
+  TooltipProps,
+  XAxis,
+  YAxis,
+  Tooltip as RechartsTooltip,
+  Area,
+  ResponsiveContainer,
+  CartesianGrid,
+} from 'recharts';
+import { Cost, DEFAULT_DATE_FORMAT, CostInsightsTheme } from '../../types';
+import {
+  BarChartTooltip as Tooltip,
+  BarChartTooltipItem as TooltipItem,
+} from '../BarChart';
+import {
+  overviewGraphTickFormatter,
+  formatGraphValue,
+  isInvalid,
+} from '../../utils/graphs';
+import {
+  useCostOverviewStyles as useStyles,
+  DataVizColors,
+} from '../../utils/styles';
+import { useFilters, useLastCompleteBillingDate } from '../../hooks';
+import { mapFiltersToProps } from './selector';
+import { getPreviousPeriodTotalCost } from '../../utils/change';
+import { LegendItem } from '../LegendItem';
+import { currencyFormatter } from '../../utils/formatters';
+import { aggregationSum } from '../../utils/sum';
+
+dayjs.extend(utc);
+
+export type CostOverviewByProductProps = {
+  dailyCostData: Cost;
+};
+
+const LOW_COST_THRESHOLD = 0.01;
+
+export const CostOverviewByProduct = ({
+  dailyCostData,
+}: CostOverviewByProductProps) => {
+  const theme = useTheme<CostInsightsTheme>();
+  const styles = useStyles(theme);
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const { duration } = useFilters(mapFiltersToProps);
+  const groupedCosts = dailyCostData.groupedCosts;
+
+  if (!groupedCosts) {
+    return null;
+  }
+
+  const flattenedAggregation = groupedCosts
+    .map(cost => cost.aggregation)
+    .flat();
+  const totalCost = flattenedAggregation.reduce(
+    (acc, agg) => acc + agg.amount,
+    0,
+  );
+
+  const productsByDate = groupedCosts.reduce((prodByDate, group) => {
+    const product = group.id;
+    group.aggregation.forEach(curAggregation => {
```

Comment:
Should avoid mutating the accumulator directly. Return a new accumulator on each iteration.